src/ebonite/client/base.py

```diff
@@ -39,10 +37,21 @@ def __init__(self, meta_repo: MetadataRepository, artifact_repo: ArtifactReposit
         self.meta_repo = meta_repo
         self.artifact_repo = artifact_repo
 
+    def delete_project(self, project, cascade=False):
+        """
+        Deletes project and(if required) all tasks associated with it from metadata repository
+        :param project: project which is meant to be deleted
+        :param cascade: whether should project be deleted with all asssociated tasks
+        :return: Nothing
+        """
+        if cascade:
+            for task in project.tasks:
```

Comment:
I recommend using `self.meta_repo.get_tasks` here as `Project.tasks` is not guaranteed to reflect DB state.